app/script/entity/message/Message.js

```diff
@@ -0,0 +1,344 @@
+
+    this.accent_color = ko.pureComputed(() => {
+      return `accent-color-${this.user().accent_id()}`;
+    });
+  }
+
+  equals(message_et) {
+    return this.id === (message_et != null ? message_et.id : undefined);
+  }
+
+  /**
+   * Check if message contains an asset of type file.
+   * @returns {Boolean} Message contains any file type asset
+   */
+  has_asset() {
+    if (this.is_content()) {
+      for (const asset_et of this.assets()) {
+        if (asset_et.type === z.assets.AssetType.FILE) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a file asset.
+   * @returns {Boolean} Message contains a file
+   */
+  has_asset_file() {
+    if (this.is_content()) {
+      for (const asset_et of this.assets()) {
+        if (asset_et.is_file()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains any image asset.
+   * @returns {Boolean} Message contains any image
+   */
+  has_asset_image() {
+    if (this.is_content()) {
+      for (const asset_et of this.assets()) {
+        if (asset_et.is_image()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a location asset.
+   * @returns {Boolean} Message contains a location
+   */
+  has_asset_location() {
+    if (this.is_content()) {
+      for (const asset_et of this.assets()) {
+        if (asset_et.is_location()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a text asset.
+   * @returns {Boolean} Message contains text
+   */
+  has_asset_text() {
+    if (this.is_content()) {
+      for (const asset_et of this.assets()) {
+        if (asset_et.is_text()) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Check if message contains a nonce.
+   * @returns {Boolean} Message contains a nonce
+   */
+  has_nonce() {
+    return this.super_type === z.message.SuperType.CONTENT;
+  }
+
+  /**
+   * Check if message is a call message.
+   * @returns {Boolean} Is message of type call
+   */
+  is_call() {
+    return this.super_type === z.message.SuperType.CALL;
+  }
+
+  /**
+   * Check if message is a content message.
+   * @returns {Boolean} Is message of type content
+   */
+  is_content() {
+    return this.super_type === z.message.SuperType.CONTENT;
+  }
+
+  /**
+   * Check if message can be deleted.
+   * @returns {Boolean} True, if message is deletable.
+   */
+  is_deletable() {
+    if (this.is_ping() || !this.has_asset()) {
+      return true;
+    }
+    return ![z.assets.AssetTransferState.DOWNLOADING,
+      z.assets.AssetTransferState.UPLOADING].includes(this.get_first_asset().status());
+  }
+
+  /**
+   * Check if the message content can be downloaded.
+   * @returns {Boolean} True, if the message has downloadable content.
+   */
+  is_downloadable() {
```

Comment:
Old code: `return @get_first_asset?().download?`

Something seems to be missing here